service-discovery/knative/src/test/java/io/smallrye/stork/servicediscovery/knative/KnativeServiceDiscoveryCDITest.java

```diff
@@ -118,7 +118,7 @@ private static Service buildKnService(String knativeService, String url, String
     @Test
     void shouldDiscoverKnativeServicesInAllNs() {
         config.addServiceConfig("my-knservice", null, "knative",
-                null, Map.of("knative-host", k8sMasterUrl, "knative-namespace", "all"));
+                null, null, Map.of("knative-host", k8sMasterUrl, "knative-namespace", "all"), null);
```

Comment:
To avoid so many changes, I would have introduced a new overloaded method, and keep the existing one available (maybe deprecated for a bit, before removal)

Reply:
Ok, atm I just added the alternative taking the registrars arguments and let both.